components/testing/cu_udpinject/src/lib.rs

```diff
@@ -0,0 +1,65 @@
+use pcap::Capture;
+use std::net::UdpSocket;
+use std::path::Path;
+use std::thread::sleep;
+use std::time::{Duration, Instant};
+
+/// Small helper to stream a pcap file over UDP
+pub struct PcapStreamer {
+    capture: Capture<pcap::Offline>,
+    socket: UdpSocket,
+    target_addr: String,
+    last_packet_ts: Option<Duration>,
+    start_instant: Instant,
+}
+
+impl PcapStreamer {
+    pub fn new(file_path: impl AsRef<Path>, target_addr: impl Into<String>) -> Self {
+        let capture = Capture::from_file(file_path).expect("Failed to open pcap file");
+        let socket = UdpSocket::bind("0.0.0.0:0").expect("Failed to bind UDP socket");
+        Self {
+            capture,
+            socket,
+            target_addr: target_addr.into(),
+            last_packet_ts: None,
+            start_instant: Instant::now(),
+        }
+    }
+
+    /// Send the next packet in the pcap file over UDP
+    /// PS is the expected payload size to send in each packet
+    /// Returns false if the end of the pcap file is reached
+    pub fn send_next<const PS: usize>(&mut self) -> bool {
+        // Get the next packet and check for end of stream
+        let packet = match self.capture.next_packet() {
+            Ok(packet) => packet,
+            Err(_) => return false, // End of the stream
+        };
+
+        // Assume 42-byte header (Ethernet + IP + UDP) and an optional 4-byte FCS
+        let payload_offset = 42;
+
+        // Extract only the payload, excluding headers and trailing FCS if present
+        let payload = &packet.data[payload_offset..payload_offset + PS];
```

Comment:
A test is missing : if the packet is smaller than expected, the slicing will panick (case occurs mix a traffic of mixed size packets)

Reply:
done